### PR TITLE
feat: add google.api_core.version_header

### PR DESCRIPTION
**Pull request by parthea**
Towards b/330968465

### CODE REVIEW

**Review by ohmayr — APPROVED** [review]

File: google/api_core/version_header.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+API_VERSION_METADATA_KEY = "x-goog-api-version"
+
+
+def to_api_version_header(version_identifier):
+    """Returns the API Version header for the given `version_identifier`.
```
Comment (vchudnov-g) [inline google/api_core/version_header.py:19]: nit: it's not the header itself, but the data for the header....

**Review by vchudnov-g — APPROVED** [review]
Thanks for doing this!

File: google/api_core/version_header.py (hunk 1 above)
Comment (parthea) [reply google/api_core/version_header.py]: Fixed in 60b33e60b795715cca9df43edd2d8ed28cd5353c

**Pull request merged**